### ISSUE TITLE
Server-side functions not working anymore in v3.2.2

### ISSUE THREAD

**Issue by Funky89** [orientechnologies/orientdb#9690]
### OS: Debian 10

Hello everyone
Having updated from version 2.2.31 to 3.2.2
I have currently done several functions but this one no longer works on 3.2.2

For example:

```
var db = orient.getGraphNoTx();

var retour = {};
odbCountryCode = odbCountryCode.toUpperCase();
var Clogins = db.getVertexType("Logins_"+ odbCountryCode) || db.createVertexType("Logins_"+ odbCountryCode, "Logins");
```
I tried by replacing orient.getGraphNoTx() by orient.getDatabase() 

`Error on parsing script at position #0: Error on execution of the script\nScript: InsertNewLogin\n------^\r\n\tDB name=\"Totoum\"\norg.graalvm.polyglot.PolyglotException: TypeError: invokeMember (getDatabase) on com.orientechnologies.orient.core.command.script.OScriptOrientWrapper@4956c6ae failed due to: Unknown identifier: getDatabase\nTypeError: invokeMember (getDatabase) on com.orientechnologies.orient.core.command.script.OScriptOrientWrapper@4956c6ae failed due to: Unknown identifier: getDatabase"`

I tried to put the value in orientdb-server-config.xml 

`<entry value="false" name="OGlobalConfiguration.SCRIPT_POLYGLOT_USE_GRAAL" />`

But in the logs here is what it returns to me:

`Ignored storage configuration because not supported: OGlobalConfiguration.SCRIPT_POLYGLOT_USE_GRAAL = null [OClusterBasedStorageConfiguration]`

I also read that the problem could come from TinkerPop which has been removed but honestly I am lost in relation to the documentation. I have not found an alternative on the functions that are no longer present
Do you have any information to provide me on the migration from the old functions to the new ones or any other information ?

Best Regards

**Comment by calebeaires** [comment]
Same here.

**Comment by calebeaires** [comment]
@Funky89 have you find a way?

**Comment by Funky89** [comment]
@calebeaires 

To deactivate GRAAL you have to use : 

`<entry name="script.polyglot.useGraal" value="false" />`

All functions have changed. So we have to replace the old ones with the new ones that are here
https://orientdb.org/docs/3.2.x/java/ref/

For info my old code:
```
var db = orient.getGraphNoTx();
var Clogins = db.getVertexType("Logins_"+ odbCountryCode) || db.createVertexType("Logins_"+ odbCountryCode, "Logins");
```
BECOMES :
```
var gdb = orient.getDatabase();
var Schema = gdb.getMetadata().getSchema ();
var Clogins = Schema.getOrCreateClass ("Logins _" + odbCountryCode, Schema.getClass ("Logins")); 
```

**Comment by tglman** [comment]
Hi All,

The use of the graal runtime for functions should be fixed from 3.2.6, with more improvements in 3.2.14 please upgrade and test again.

Bye

**Comment by praveen-mamdge** [comment]
@tglman I am encountering an exception when attempting to use the getDatabase() function in OrientDB Studio with GraalVM enabled. When I execute the script, 
`let db = orient.getDatabase()`

The following exception I see: 
`{
    "errors": [
        {
            "code": 400,
            "reason": "Bad request",
            "content": "Error on parsing script at position #10: Error on execution of the script\nScript: TestFunction\n---------------^\r\n\tDB name=\"TestDB\"\norg.graalvm.polyglot.PolyglotException: TypeError: invokeMember (getDatabase) on null failed due to: Message not supported.\nTypeError: invokeMember (getDatabase) on null failed due to: Message not supported."
        }
    ]
}`

However, when I disable GraalVM using the script.polyglot.useGraal = false option, then I don't get any exception.

I am using OrientDB version 3.2.14. Is it possible to use server-side functions with GraalVM, and if so, how can I resolve the error I am experiencing when using getDatabase()? 

Please let me know if you need any other information.

**Comment by tglman** [comment]
Hi,

the reason for this error is that the method `orient.getDatabase()` to get the database from a function has been deprecated since a while now is enough just use the variable `db` that should work with both engines.

Regards

**Comment by praveen-mamdge** [comment]
@tglman Thank you, server side script works without any issues.

**Comment by tglman** [comment]
Ok, so can I close this issue ?

**Comment by praveen-mamdge** [comment]
@tglman The scripts gets executed without any issues. Whereas we are facing issue while accessing the data received the query result - 

When we execute the command `var data = db.command("select from #249:0")`, we are not able to access the result, even the object with the given RID is exist but when we try to access the data by logging it on console via stringify it is empty object. If we just log the received record then we get the result as "com.orientechnologies.orient.core.sql.parser.OLocalResultSetLifecycleDecorator@1efe660b" on the console.

Is there any handling we need to do explicitly to access the data received after running the command?

E.g. If we run the following script by disabling the UseGraal setting, we are getting JSON object with the details of record.
```
var db = orient.getDatabase();
var result = db.command("select from #249:0");
var data = result[0];
return data
```

Whereas if we run the same script with only change of replacing orient.getDatabase with db variable and enabling the UseGraal setting, we are getting undefined or null result - 
```
var result = db.command("select from #249:0");
var data = result[0];
return data
```

Please let me know if you need any additional details.

**Comment by praveen-mamdge** [comment]
Hi @tglman, will you please provide your inputs on above? If you have need any other details then please let me know.

**Comment by tglman** [comment]
Hi @praveen-mamdge ,

So the command seems to return a result set, so you could return it directly or do:
```
var result = db.command("select from #249:0");
var data = result.next();
result.close();
return data
```
Regards

**Comment by praveen-mamdge** [comment]
Thanks @tglman.

There is one observation, now on db object the **commit** and **rollback** methods are not available? Ater enabling the graalVM we can see error for db.commit and db.rollback methods. We are getting following error.

`
{
    "errors": [
        {
            "code": 400,
            "reason": "Bad request",
            "content": "Error on parsing script at position #1: Error on execution of the script\nScript: TestFunction\n------^\r\n\tDB name=\"MC_InteliNotionDB\"\norg.graalvm.polyglot.PolyglotException: TypeError: invokeMember (rollback) on com.orientechnologies.orient.core.command.script.OScriptDatabaseWrapper@536ebdbf failed due to: Unknown identifier: rollback\nTypeError: invokeMember (rollback) on com.orientechnologies.orient.core.command.script.OScriptDatabaseWrapper@536ebdbf failed due to: Unknown identifier: rollback"
        }
    ]
}
`

**Comment by tglman** [comment]
Hi,

this should be already fixed in 3.2.22, closing, feel free to reopen if you still have problems.

Regards
